### PR TITLE
Terraform Parser

### PR DESCRIPTION
**Pull request by joaotgoncalves**
Created the parser for Terraform files. In this pull request is also included the merge from IR Extension PR, to have the desired intermediate representation. The actual version of the parser is parsing Terraform resources, data sources, input variables, local values, output values, module blocks and comments.

The following example shows some terraform code and its corresponding parser

```
resource "aws_instance" "demo" {
    ami = "ami-00831fc7c1e3ddc60"
    instance_type = "t2.micro"
    Tags {
        Name = "ExampleEc2Instance"
    } 
}
```

Parsed example:

examplePR

### CODE REVIEW

**Review by Nfsaavedra — CHANGES_REQUESTED** [review]
Besides the single comment I did, the code looks good to me. However, we should have integration tests for Terraform for each of the smells already implemented. Also, it would be nice to add unit tests for some edge cases that you found in the parsing (it is not done for the other technologies but should be considered in the future) -- I do not consider the unit tests urgent, but if it is simple please do it. Also, please update the README to add the Terraform technology, namely, this line:

```
glitch --tech (chef|puppet|ansible) --csv --config PATH_TO_CONFIG PATH_TO_FILE_OR_FOLDER
```

File: glitch/tests/parser/terraform/files/comments.tf [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#comment1
+//comment2
+resource "google_bigquery_dataset" "dataset" {
```
Comment (Nfsaavedra) [inline glitch/tests/parser/terraform/files/comments.tf:3]: To keep the consistency to the style conventions of Terraform this file should  also have 2 space identation

**Review by Nfsaavedra — APPROVED** [review]
LGTM, very nice job :) The test set seems very complete. Please just change the indentation in the `comments.tf` file! We should wait for @jff review and if the code looks good to him as well, we can merge.

File: requirements.txt [hunk 2]
```diff
@@ -7,4 +7,5 @@ pandas
 configparser
 pytest
 puppetparser
-jinja2
\ No newline at end of file
+jinja2
+glitch-python-hcl2
```
Comment (jff) [inline requirements.txt:11]: This is fine, but it might be a good idea to include versions (perhaps the best is to create a separate issue for that). Otherwise, we might have issues in a near future related to incompatible versions.

**Comment by jff** [comment]
LGTM! Thanks a lot for this! :clap:

**Review by jff — APPROVED** [review]

File: requirements.txt (hunk 2 above)
Comment (Nfsaavedra) [reply requirements.txt:11]: I will open a issue for that
